x/profiles/types/msgs/msgs.go

```diff
@@ -152,61 +152,153 @@ func (msg MsgRequestDTagTransfer) GetSignBytes() []byte {
+	return []sdk.AccAddress{msg.Receiver}
+}
+
+// MsgRefuseDTagTransferRequest represent a DTag request rejection
+type MsgRefuseDTagTransferRequest struct {
+	Receiver sdk.AccAddress `json:"receiver" yaml:"receiver"`
+	Sender   sdk.AccAddress `json:"sender" yaml:"sender"`
+}
+
+// NewMsgRefuseDTagTransferRequest is a constructor for MsgRefuseDTagTransferRequest
+func NewMsgRefuseDTagTransferRequest(receiver, sender sdk.AccAddress) MsgRefuseDTagTransferRequest {
+	return MsgRefuseDTagTransferRequest{
+		Receiver: receiver,
+		Sender:   sender,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgRefuseDTagTransferRequest) Route() string { return models.RouterKey }
+
+// Type should return the action
+func (msg MsgRefuseDTagTransferRequest) Type() string { return models.RefuseDTagTransferRequest }
+
+func (msg MsgRefuseDTagTransferRequest) ValidateBasic() error {
+	if msg.Sender.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid sender address: %s", msg.Sender))
+	}
+
+	if msg.Receiver.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid receiver address: %s", msg.Receiver))
+	}
+
+	if msg.Sender.Equals(msg.Receiver) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "the sender and receiver addresses must be different")
+	}
+
+	return nil
+}
+
+// GetSignBytes encodes the message for signing
+func (msg MsgRefuseDTagTransferRequest) GetSignBytes() []byte {
+	return sdk.MustSortJSON(MsgsCodec.MustMarshalJSON(msg))
+}
+
+// GetSigners defines whose signature is required
+func (msg MsgRefuseDTagTransferRequest) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// MsgCancelDTagTransferRequest represent a DTag request rejection
+type MsgCancelDTagTransferRequest struct {
+	Sender   sdk.AccAddress `json:"sender" yaml:"sender"`
+	Receiver sdk.AccAddress `json:"receiver" yaml:"receiver"`
+}
+
+// NewMsgCancelDTagTransferRequest is a constructor for MsgCancelDTagTransferRequest
+func NewMsgCancelDTagTransferRequest(sender, receiver sdk.AccAddress) MsgCancelDTagTransferRequest {
+	return MsgCancelDTagTransferRequest{
+		Sender:   sender,
+		Receiver: receiver,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgCancelDTagTransferRequest) Route() string { return models.RouterKey }
+
+// Type should return the action
+func (msg MsgCancelDTagTransferRequest) Type() string { return models.CancelDTagTransferRequest }
+
+func (msg MsgCancelDTagTransferRequest) ValidateBasic() error {
+	if msg.Receiver.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid receiver address: %s", msg.Receiver))
+	}
+
+	if msg.Sender.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid sender address: %s", msg.Sender))
+	}
+
+	if msg.Receiver.Equals(msg.Sender) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "the receiver and sender addresses must be different")
```

Comment:
Again, let's change it to be more coherent

```suggestion
		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "the receiver and sender must be different")
```